### PR TITLE
Button component

### PR DESCRIPTION
**Pull request by DanWebb**

### CODE REVIEW

File: components/control/Button/Button.css [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+
+/**
+ * diamond-button specific styles
+ */
+diamond-button a {
+  color: var(--_color);
+  text-decoration: none;
+}
+
+diamond-button :where(button),
```
Comment (gavmck) [inline components/control/Button/Button.css:123]: Why where and not a child selector?
You can do multiple wheres

`diamond-button :where(button, input)`

File: components/control/Button/Button.css [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+diamond-button :where(input[type='submit']),
+diamond-button :where(input[type='reset']),
+diamond-button :where(input[type='file']),
+diamond-button :where(input[type='file'])::file-selector-button,
+diamond-button a {
+  diamond-button[size='sm'] & {
+    --_size: 0.875rem;
+  }
+
+  diamond-button[size='lg'] & {
```
Comment (gavmck) [inline components/control/Button/Button.css:134]: Does this need the el prefix if we're already in an el selector?

File: components/control/Button/Button.css [hunk 3]
```diff
@@ -0,0 +1,195 @@
...
+    --_size: 1.25rem;
+  }
+
+  diamond-button[width='full-width'] & {
+    width: 100%;
+  }
+
+  @media (max-width: 480px) {
```
Comment (gavmck) [inline components/control/Button/Button.css:142]:
```suggestion
  @media (width < 480px) {
```

480 includes 480 so all the min width ones will flip at 480, but this one won't. Relative media queries are less ambiguous

File: components/control/Button/Button.css [hunk 4]
```diff
@@ -0,0 +1,195 @@
...
+  diamond-button[size='sm'] & {
+    --_size: 0.875rem;
+  }
+
+  diamond-button[size='lg'] & {
+    --_size: 1.25rem;
+  }
+
+  diamond-button[width='full-width'] & {
+    width: 100%;
```
Comment (gavmck) [inline components/control/Button/Button.css:139]: Do these need display block to work

File: components/control/Button/Button.stories.ts [hunk 5]
```diff
@@ -0,0 +1,238 @@
...
+      </diamond-grid-item>
+    </diamond-grid>
+  `,
+};
+
+export const IconOnlyButton: StoryObj = {
+  render: () => html`
+    <diamond-grid wrap="wrap">
+      <diamond-grid-item>
+        <diamond-button width="square">
```
Comment (gavmck) [inline components/control/Button/Button.stories.ts:111]: Should we be explicit and say width icon?

File: components/control/Button/Button.css [hunk 6]
```diff
@@ -0,0 +1,195 @@
+/**
+ * Generic styles for all buttons
+ */
+button,
```
[the remark this answers is not in the document]
Comment (DanWebb) [reply components/control/Button/Button.css:4]: @gavmck I think all the possible buttons are showing in the HTMLButton story I've added.
If we wanted to be safe I could use the diamond-button selector on top of everything else, it means every button would have to be wrapped in the component to get the styles.

File: components/control/Button/Button.css [hunk 7]
```diff
@@ -0,0 +1,195 @@
...
+diamond-button a,
+input[type='button'],
```
[the remark this answers is not in the document]
Comment (DanWebb) [reply components/control/Button/Button.css:6]: Same as https://github.com/etchteam/diamond-ui/pull/12#discussion_r1478454009

File: components/control/Button/Button.css (hunk 1 above)
Comment (DanWebb) [reply components/control/Button/Button.css:123]: It overwrites without :where

File: components/control/Button/Button.css (hunk 2 above)
Comment (DanWebb) [reply components/control/Button/Button.css:134]: it's selecting the child button/input/a not the component

File: components/control/Button/Button.css (hunk 4 above)
Comment (DanWebb) [reply components/control/Button/Button.css:139]: No, see the story.

File: components/control/Button/Button.stories.ts (hunk 5 above)
Comment (DanWebb) [reply components/control/Button/Button.stories.ts:111]: I don't think we need to tie it to an icon. Width icon could do anything, square is a shape everyone knows

File: components/control/Button/Button.css (hunk 6 above)
[the remark this answers is not in the document]
Comment (gavmck) [reply components/control/Button/Button.css:4]: Yeah, that was the only thing I was questioning. Do we want it to blanket apply to all buttons? Don't mind if we do

[the remark this answers is not in the document]
Comment (DanWebb) [reply components/control/Button/Button.css:4]: I found it applying styles to the button card so I've scoped it to diamond-button

**Review by gavmck — APPROVED** [review]
🔥

**Pull request merged**